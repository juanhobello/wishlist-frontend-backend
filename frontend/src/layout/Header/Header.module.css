.header {
  background-color: var(--primary-color);
  color: var(--white-color);
  display: grid;
  place-items: center;
  height: 6rem;
  position: sticky;
  top: 0;
  z-index: 999;
}

.header-content {
  width: calc(100% - 4rem);
  max-width: 80rem;
  height: calc(6rem - 4rem);
  margin: 0 auto;
  display: inline-flex;
  justify-content: space-between;
  align-items: center;
  padding: 2rem;
  gap: 2rem;
}

.first-section {
  width: 10rem;
  height: 100%; 
}

.second-section {
  width: 100%; 
  max-width: 10rem;
  display: flex;
  justify-content: space-between;
}
