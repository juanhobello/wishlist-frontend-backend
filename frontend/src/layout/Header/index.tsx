import { CustomLink, Logo, Menu, MenuItem } from '@/components'
import { FiHeart } from "react-icons/fi";
import { FaUserCircle } from "react-icons/fa";
import style from './Header.module.css'
import { useMenu } from '@/hooks';

export default function Header() {
  const { open, anchorEl, handleOpen, handleClose } = useMenu();

  return (
    <header className={style["header"]}>
      <div className={style["header-content"]}>
        <section className={style['first-section']}>
          <Logo path="/" />
        </section>

        <section className={style['second-section']}>
          <CustomLink path="/wishlist"><FiHeart fontSize={24} /><span>Wishlist</span></CustomLink>

          <span onClick={handleOpen} style={{ cursor: 'pointer' }}><FaUserCircle fontSize={24} /></span>

          <Menu open={open} archorEl={anchorEl} onClose={handleClose}>
            <MenuItem onClick={handleClose}>Entrar</MenuItem>
            <MenuItem onClick={handleClose}>Minha Conta</MenuItem>
            <MenuItem onClick={handleClose}>Endere√ßos</MenuItem>
            <MenuItem onClick={handleClose}>Minha Netshoes</MenuItem>
          </Menu>
        </section>
      </div>
    </header>
  )
}
