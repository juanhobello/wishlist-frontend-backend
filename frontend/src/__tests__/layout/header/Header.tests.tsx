import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { BrowserRouter } from 'react-router-dom';
import Header from "@/layout/Header"

describe('Header component', () => {
  test("matches snapshot", () => {
    const { container } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });

  test("Header should render correctly", async () => {
    render(<BrowserRouter><Header /></BrowserRouter>);
    expect(screen.getByRole("link", { name: /Wishlist/i })).toBeInTheDocument();
    expect(screen.getByRole("link", { name: /Logo/i })).toBeInTheDocument();
    expect(screen.getByRole("button", { name: /User/i })).toBeInTheDocument();
  });
})