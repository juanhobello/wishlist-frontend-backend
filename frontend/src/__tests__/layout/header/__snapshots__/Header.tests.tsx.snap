// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="header-content"
    >
      <section
        class="first-section"
      >
        <a
          aria-label="Logo"
          class="logo-container"
          data-discover="true"
          href="/"
        >
          <div
            class="logo"
          />
        </a>
      </section>
      <section
        class="second-section"
      >
        <a
          class="link-wrapper"
          data-discover="true"
          href="/wishlist"
        >
          <svg
            fill="none"
            font-size="24"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20.84 4.61a5.5 5.5 0 0 0-7.78 0L12 5.67l-1.06-1.06a5.5 5.5 0 0 0-7.78 7.78l1.06 1.06L12 21.23l7.78-7.78 1.06-1.06a5.5 5.5 0 0 0 0-7.78z"
            />
          </svg>
          <span>
            Wishlist
          </span>
        </a>
        <span
          aria-label="User"
          role="button"
          style="cursor: pointer;"
        >
          <svg
            fill="currentColor"
            font-size="28"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 496 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
            />
          </svg>
        </span>
      </section>
    </div>
  </header>
</div>
`;
